### PR TITLE
Pivotal ID #185392780: User FTP files API support

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/185392780
- added ftp magic folder support

### CODE REVIEW

File: submission/submission-security/src/main/kotlin/ebi/ac/uk/security/service/SecurityService.kt [hunk 1]
```diff
@@ -152,23 +149,17 @@ open class SecurityService(
...
-        return Paths.get("${securityProps.magicDirPath}/$prefixFolder/$userEmail")
-    }
-
     private fun asUser(registerRequest: RegisterRequest) = DbUser(
         email = registerRequest.email,
         fullName = registerRequest.name,
         orcid = registerRequest.orcid,
         secret = securityUtil.newKey(),
         notificationsEnabled = registerRequest.notificationsEnabled,
+        magicFolderType = MagicFolderType.NFS,
```
Comment (awaisathar) [inline submission/submission-security/src/main/kotlin/ebi/ac/uk/security/service/SecurityService.kt:162]: Would be great to have the default value configurable in application properties

**Comment by Juan-EBI** [comment]
@jhoanmanuelms  after this one we will need to
1. Update user table to include new column
2. Update properties to have new organization

File: submission/submission-webapp/src/itest/resources/application.yml [hunk 2]
```diff
@@ -43,9 +43,11 @@ app:
   security:
     tokenHash: biostd-test
     environment: TEST
-    filesDirPath:
-    magicDirPath:
     requireActivation: false
+    filesProperties:
+      filesDirPath:
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/itest/resources/application.yml:48]: I think we could use this opportunity to add documentation for the properties

**Review by jhoanmanuelms — APPROVED** [review]
LGTM. Please consider my comments

File: submission/submission-webapp/src/itest/resources/application.yml (hunk 2 above)
Comment (Juan-EBI) [reply submission/submission-webapp/src/itest/resources/application.yml:48]: they have, but not in the itest file but in application yml files of reulgar application

File: submission/submission-security/src/main/kotlin/ebi/ac/uk/security/service/SecurityService.kt [hunk 3]
```diff
@@ -169,6 +182,7 @@ open class SecurityService(
         orcid = registerRequest.orcid,
         secret = securityUtil.newKey(),
         notificationsEnabled = registerRequest.notificationsEnabled,
+        magicFolderType = securityProps.filesProperties.defaultMode,
```
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/submission-security/src/main/kotlin/ebi/ac/uk/security/service/SecurityService.kt]: to come in following PR

**Pull request merged**